src/ng/compile.js

```diff
@@ -2480,7 +2480,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
 
               // We have transclusion slots,
               // collect them up, compile them and store their transclusion functions
-              $template = [];
+              $template = window.document.createDocumentFragment();
```

Comment:
Other places inside this file use `window` directly (e.g. [here][1], [here][2] and [here][3]). Normally, I would prefer `$window` (or `$document`). But it doesn't bother me too much here because:

1. This file is dealing with pretty "low-level" stuff.
2. Switching to `$window` would be a breaking change (since every `$window` or `$document` mock should then provide the necessary APIs).

[1]: https://github.com/gkalpak/angular.js/blob/5a2db25b5d5d331f414894f3e13c096c8ec85779/src/ng/compile.js#L1495
[2]: https://github.com/gkalpak/angular.js/blob/5a2db25b5d5d331f414894f3e13c096c8ec85779/src/ng/compile.js#L1840
[3]: https://github.com/gkalpak/angular.js/blob/5a2db25b5d5d331f414894f3e13c096c8ec85779/src/ng/compile.js#L3389